src/devSVG.cpp

```diff
@@ -429,10 +495,25 @@ void svg_path(double *x, double *y,
 double svg_strwidth(const char *str, const pGEcontext gc, pDevDesc dd) {
   SVGDesc *svgd = (SVGDesc*) dd->deviceSpecific;
 
-  gdtools::context_set_font(svgd->cc, fontname(gc->fontfamily, gc->fontface),
-    gc->cex * gc->ps, is_bold(gc->fontface), is_italic(gc->fontface));
+  std::string file = fontfile(gc->fontfamily, gc->fontface, svgd->font_aliases, svgd->font_extra);
+  gdtools::context_set_font(svgd->cc, fontname(gc->fontfamily, gc->fontface, svgd->font_aliases),
+                            gc->cex * gc->ps, is_bold(gc->fontface), is_italic(gc->fontface), file);
   FontMetric fm = gdtools::context_extents(svgd->cc, std::string(str));
 
+  if (fm.width == 0) {
+    int length = 0;
+    if (mbcslocale) {
+      // Compute width of UTF8 string
+      const char* s = str;
+      while (*s)
```

Comment:
This is counting the number of UTF8 separators. After seeing your other link I realise that mbcs is about many more encodings than UTF8. So I wrapped `str` in a CHARSXP and used `R_nchar()`, thanks for the pointer.

This somehow still fails on Travis, I'm setting up a Ubuntu virtualbox to investigate.